CONTRIBUTING.md

```diff
@@ -0,0 +1,23 @@
+## Contributing to this repository
+
+1. Don't edit the Dockerfiles directly. They are generated using templates.
+2. Make any changes to the `*.template` files in the root of the repository.
+3. Make sure you've checked the [Requirements](#requirements) below.
+4. Run the `./apply-templates.sh` script after making your changes.
+5. Once complete, commit all changes to the templates and Dockerfiles.
+
+## Requirements
+
+You will need [Gnu Awk](https://www.gnu.org/software/gawk/) to generate the Dockerfiles from the templates.
```

Comment:
This is a great start, but it's unfortunately also going to require both `jq` and reasonably recent `bash` (the latter probably only relevant to mention here for MacOS users).

Perhaps we instead just list the requirements here and leave "how to install" as an exercise for the reader, since preferences there will likely vary a bit more wildly?